lipx.py

```diff
@@ -45,6 +35,17 @@ def get_uint24(data, index):
     return int((data[index] << 16) | (data[index + 1] << 8) | data[index + 2])
 
 
+# Helper function to display patch data in INSPECT mode
+def format_patch(offset, size, data, rle=0):
+    if (offset ^ size) & 1:
```

Comment:
any particular reason why you are doing this?

this is true in some cases when size == 0 and it crashes

![Screenshot_20230711_190124](https://github.com/kylon/Lipx/assets/3252255/a7ea2a64-f1d6-4138-b2e4-5840d217172a)

```suggestion
    if size:
```

Reply:
... I think this was me trying to be clever and making that odd / even offset idea mentioned above work.  It appeared to work on my data, but there's obviously some case I've missed. I'll work on it.

Reply:
you can reproduce with https://www.romhacking.net/hacks/6373/

Reply:
Thanks, those are good test examples. I hadn't really considered long patches (100 byte plus) and line wrapping.
I've fixed the bug, but the long patches still look ugly.

Reply:
@kylon The latest commit dfe856e256cef78d1fe6871e9a4aab9b24e58291 displays the bigger patches like this now:

```
0224e0   (16): 00   <REPEAT x 16>
0224f0  (256): 0301 0000 0000 0000 1c06 0100 0000 0000
               d5ea 753e 0f03 213c 0000 80c0 701c 263f
               0102 0102 0102 0102 0000 0000 0000 0000
               60e0 60e0 40c0 c0c0 1010 1010 3020 2020
               00f3 0efb fbff fb9b 00f3 ffff ffff ffff
               1e0e 0e0c 0404 0400 2131 1113 1a0a 0a0e
               0000 0000 0000 0000 0007 0d0a 0c0c 0600
               0000 0000 001f 1f00 0000 0000 001f 1f00
               1d1a 0d2e 2fef e717 e020 3030 30f0 f818
               70e0 e3c3 c393 bb3b 0818 1333 2373 7bfb
               0000 c0ff fff0 f0f0 0000 c0ff fff0 f0f0
               fcfc fcff fffc fcfc fcfc fcff fffc fcfc
               0726 77f6 f33b 3b3b 1838 78f8 fc3c 3c3c
               54ac 5cac 5cac 5cac 0202 0202 0202 0202
               0404 0404 0000 0000 0a0a 0a0a 0606 0602
               0000 0000 0000 0003 0000 0000 0406 0704
0225f0   (14): 00   <REPEAT x 14>
0225fe    (2): 80c0
022600   (16): 3f   <REPEAT x 16>
022610   (58): 0502 0502 0902 0502 0000 0000 0000 0000
               40c0 c0c0 c0c0 c080 2020 2020 2020 2060
               eb9b 1b3b 7bff f300 ffff ffff ffff f300
               0000 0000 0000 0000 0404
02264a   (15): 00   <REPEAT x 15>
022659    (6): ..f0 d828 9818 30
02265f   (17): ..00 <REPEAT x 17>
022670  (144): 1716 3238 3979 7dfc 1819 3d3f 3f7f 7fff
               7b73 e3c3 c383 83f3 fbf3 e3c3 c383 83f3
               f0f0 f0f0 f0f0 f0f1 f0f0 f0f0 f0f0 f1f2
```

It might make sense to add a display of any printable bytes in a column to the right. I'm not sure; most of the data I'm looking at isn't ASCII. I'm going to focus on the other refactoring next, and we can tweak the display output later if needed.